src/effects/VST/VSTEffect.cpp

```diff
@@ -2147,6 +2193,16 @@ intptr_t VSTEffectWrapper::callDispatcher(int opcode,
    return mAEffect->dispatcher(mAEffect, opcode, index, value, ptr, opt);
 }
 
+intptr_t VSTEffectWrapper::callDispatcher(
+   AEffect* handle,
+   int opcode, int index, intptr_t value, void* ptr, float opt)
+{
+   // Needed since we might be in the dispatcher when the timer pops
+   wxCRIT_SECT_LOCKER(locker, mDispatcherLock);
```

Comment:
If this function is redefined as `static`, which I recommend, then `mDispatcherLock` must become an argument, passed by reference.

And you can also modernize by using `std::mutex` instead of the wx synchronization classes.

Reply:
Here's a better idea for splitting Load, phase 2.

Define a small structure holding mAEffect and the mutex.  Make `callDispatcher` a member function of it that locks the mutex.  Make its destructor call the dispatcher with `effClose`, replacing the two places where you do that explicitly.

You can also change the existing `mAEffect` member to be of this structure type, and define an `operator ->`, which will let you avoid changing many lines of code.